### PR TITLE
Add Secret Manager IAM samples

### PR DESCRIPTION
**Pull request by sethvargo**

### CODE REVIEW

File: secretmanager/src/main/java/com/example/IamRevokeAccess.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      // Create a request to get the current IAM policy.
+      GetIamPolicyRequest getRequest =
+          GetIamPolicyRequest.newBuilder().setResource(name.toString()).build();
+
+      // Request the current IAM policy.
+      Policy policy = client.getIamPolicy(getRequest);
+
+      // Search through bindings and remove matches.
+      String roleToFind = "roles/secretmanager.secretAccessor";
+      for (Binding b : policy.getBindingsList()) {
```
Comment (averikitsch) [inline secretmanager/src/main/java/com/example/IamRevokeAccess.java]: nit: better variable name for `b`

File: secretmanager/src/test/java/com/example/SnippetsIT.java [hunk 2]
```diff
@@ -47,6 +47,7 @@
 /** Integration (system) tests for {@link Snippets}. */
 @RunWith(JUnit4.class)
 public class SnippetsIT {
+  private static final String IAM_USER = "user:sethvargo@google.com";
```
Comment (averikitsch) [inline secretmanager/src/test/java/com/example/SnippetsIT.java]: probably should change this

Comment (sethvargo) [reply secretmanager/src/test/java/com/example/SnippetsIT.java]: Hi @averikitsch - is there a good service account that exists (or could you create one) in the test project please?

File: secretmanager/src/main/java/com/example/IamRevokeAccess.java (hunk 1 above)
Comment (sethvargo) [reply secretmanager/src/main/java/com/example/IamRevokeAccess.java]: Done

File: secretmanager/src/test/java/com/example/SnippetsIT.java (hunk 2 above)
Comment (averikitsch) [reply secretmanager/src/test/java/com/example/SnippetsIT.java]: iam-samples@java-docs-samples-testing.iam.gserviceaccount.com. Currently it has no roles or keys associated with it.

Comment (sethvargo) [reply secretmanager/src/test/java/com/example/SnippetsIT.java]: Done

**Comment by sethvargo** [comment]
@averikitsch thanks for catching the prefix 😄 - this LGTM. Do you want to merge or should i?

**Pull request merged**